CONTRIBUTING.md

```diff
@@ -44,6 +44,14 @@ In this case the reviewer or PR maintainer needs to make sure to create an issue
 
 Find here [a guide](doc/setup/SETUP.md) on how to setup your local development environment.
 
+## Route Naming Conventions
+
+- Always use **kebab-case** (e.g. 'exampleAssessment' → 'example-assessment')
+- The routes should follow the general structure entity > entityId > subentity ... (e.g. '/exercises/{exerciseId}/participations')
+- Use **plural for server route's** entities and **singular for client route's** entities
+- Specifiy the key entity at the end of the route (e.g. text-editor/participations/{participationId} → participations/{participationId}/text-editor)
+- Never specify an id that is used only for consistency and not used in the code
```

Comment:
```suggestion
- Never specify an id that is used only for consistency and not used in the code (e.g. GET "/courses/{courseId}/exercises{exerciseId}/participations/{participationId}/submissions/{submissionId}" can be simplified to GET "/submissions/{submissionId}" because all other entities than the submission are either not needed or can be loaded without the need to specify the id)
```